src/classes/RD2_RecurringDonationsGateway.cls

```diff
@@ -0,0 +1,131 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Recurring Donations Gateway to return data
+*
+*/
+public with sharing class RD2_RecurringDonationsGateway {
+
+    /***
+    * @description Retrieves Recurring Donation records for specified Account Ids
+    * @param accountIds Account Ids
+    * @return List<npe03__Recurring_Donation__c>
+    */
+    public List<npe03__Recurring_Donation__c> getRecordsReferencingAccounts(Set<Id> accountIds) {
```

Comment:
I'm not necessarily seeing these method used outside of a unit test. Are these left overs from the renaming/merging logic that was removed?

Out of curiosity, could these two methods be merged into a single `getRecordsForAccountOrContact()` method that still accepts `Set<Id>` and filters for `npe03__Organization__c IN :parentIds OR npe03__Contact__c IN :parentIds`? Mainly I'm worried about both methods being called in the same execution context (i.e., two queries). Plus the methods are very similar.

Reply:
These left overs from the renaming/merging logic that was removed. Since RD retrieval as of current WIs in progress is applied in unit tests, I've created `TEST_SObjectGateway` to be used to retrieve data in tests. (Sorry for multiple comment emails, some comments are not visible In-Review mode).